test/cbmc/proofs/UDP/vProcessGeneratedUDPPacket/vProcessGeneratedUDPPacket_harness.c

```diff
@@ -91,5 +130,44 @@ void harness()
     __CPROVER_assume( pxNetworkBuffer != NULL );
     __CPROVER_assume( pxNetworkBuffer->pucEthernetBuffer != NULL );
 
+    /* Arbitrarily add endpoint to the network buffer. This can be NULL or
+     * a proper end point. */
+    pxNetworkBuffer->pxEndPoint = nondet_bool() ? NULL : &xLocalEndPoint;
+
+    /* Assume that the list of interfaces/endpoints is not initialized.
+     * These are defined in the FreeRTOS_Routing.c file and used as a
+     * list by the TCP stack. */
+    __CPROVER_assume( pxNetworkInterfaces == NULL );
+    __CPROVER_assume( pxNetworkEndPoints == NULL );
+
+    /* Non-deterministically add a network-interface. */
+    if( nondet_bool() )
+    {
+        /* Add the network interfaces to the list. */
+        FreeRTOS_AddNetworkInterface( &xNetworkInterface1 );
+
+        /* Non-deterministically add an end-point to the network-interface. */
+        if( nondet_bool() )
+        {
+            /* Fill the endpoints and put them in the network interface. */
+            FreeRTOS_FillEndPoint( &xNetworkInterface1,
+                                   &xEndPoint1,
+                                   ucIPAddress2,
+                                   ucNetMask2,
+                                   ucGatewayAddress2,
+                                   ucDNSServerAddress2,
+                                   ucMACAddress );
+
+            /* Add the output function to the endpoint. */
+            xEndPoint1.pxNetworkInterface = nondet_bool() ? NULL : malloc( sizeof( *( xEndPoint1.pxNetworkInterface ) ) );
+
+            /* If a network interface has an endpoint, then it must have an output function. */
+            __CPROVER_assume( xEndPoint1.pxNetworkInterface != NULL );
```

Comment:
It is a little odd to use the nondet_bool() expression to let the pointer be null, and then immediately say the point is not null.    In these cases, I would use straight malloc, but I would retain the nonnull assumption in case someone someday runs cbmc with the --malloc-can-fail flags.  If you do this here, consider doing it on all PRs#180-185.

Reply:
I have added that in anticipation of using the `--malloc-can-fail` flag. But when I just use a malloc, the use of `__CPROVER_assume` seems weird.
I am thinking to add the flag and clean up all the proofs all at one with an additional PR.